### DISCUSSION TITLE
Could Python fetch... methods return tuple type?

### DISCUSSION THREAD

**Discussion by davetapley** [duckdb/duckdb#5729] [Ideas]
This all type checks fine ✅ 

```py
import duckdb

con = duckdb.connect()
con.execute('''
create table test (foo text, bar int);
insert into test values ('one', 1);
insert into test values ('two', 2);
''')

rows = con.execute('select foo, bar from test').fetchmany()
for row in rows:
    foo, bar = row
    print(foo, bar + 1)
```

But this does not ❌ 

```python
row_one = con.execute('select foo, bar from test').fetchone()
foo, bar = row_one
print(foo, bar + 1)

# pyright:
#   "object" is not iterable
#     "__iter__" method not definedPylance[reportGeneralTypeIssues](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportGeneralTypeIssues)

# mypy:
#   error: Value of type "object" is not indexable  [index]
#   Found 1 error in 1 file (checked 1 source file)
```

---

`fetchmany` only type checks because `row` is `Unknown` since the collection is `list`:
https://github.com/duckdb/duckdb/blob/5d86301ac6ded9d84073ffa95cf0c1f177991f3f/tools/pythonpkg/duckdb-stubs/__init__.pyi#L61

`fetchone` reveals the true problem, since [stubgen](https://mypy.readthedocs.io/en/stable/stubgen.html) gives it `object`: 

https://github.com/duckdb/duckdb/blob/5d86301ac6ded9d84073ffa95cf0c1f177991f3f/tools/pythonpkg/duckdb-stubs/__init__.pyi#L65

---

I'd argue that [`tuple`](https://docs.python.org/3/library/typing.html#typing.TypeVarTuple) would be more appropriate in both cases. The values themselves would still be `Unknown`, but at least unpacking from `fetchone` would type check.

I could have a go at a PR, but I'm guessing it'd be non-trivial since the types [come from stubgen](https://github.com/duckdb/duckdb/blob/5d86301ac6ded9d84073ffa95cf0c1f177991f3f/scripts/regenerate_python_stubs.sh)? 😬

**Comment by Tishj** [comment]
Already fixed, the return type of `fetchone` is now `typing.Optional[tuple]`

**Reply by davetapley** [reply]
Yes! Shame GH didn't link PR 🤔 

https://github.com/duckdb/duckdb/pull/5732
